### ISSUE TITLE
Part 1  - Feedback

### ISSUE THREAD

**Issue by jrocaspana** [gianpieryup/MakingSense#1]
Hi @gianpieryup !  excellent work!! Please find below a list with some things I believe we can improve. 

- let's code in english!
- let's try to manipulate a collection of cars instead of the JSON file on each method (meaning not having to deal with the index of the element). As an example: you can possibly load all the cars in memory when the CRUD class is instantiated and persist the changes on demand.
- let's make the Car class have its own file (not inside Program.cs)
- instead of using an INT to indicate whether a car is manual or not, let's try to use an enum.
- try to use implicit types when possible (remember to follow code conventions)

feel free to push the changes in different branches and opening pull requests so we can review them more in detail and clarify any doubt you may have.

**Comment by gianpieryup** [comment]
![image](https://user-images.githubusercontent.com/37640669/131752059-6123832a-5a0b-4b7c-88d4-808eff707ab4.png)

**Comment by gianpieryup** [comment]
Ya pude
![image](https://user-images.githubusercontent.com/37640669/131753807-93487eff-a3e6-4a09-af29-709418f6482a.png)

**Comment by gianpieryup** [comment]
@jrocaspana 
Lo que me pasa ahora es que no me reconoce el tipo de dato creado, a pesar de estar en el mismo namespace
![image](https://user-images.githubusercontent.com/37640669/131753891-36993482-ebf8-4cbc-864a-020ed11d6484.png)

**Comment by gianpieryup** [comment]
Lo arregle era  
`Car carNew = new Car("Ford", "K21", 4, "ROJO", type.Manual); `
Ya no tengo errores pero lo que pasa es que ese campo no se guarda en el JSON

**Comment by jrocaspana** [comment]
hi @gianpieryup, let me try to answer below:

- `type` is a reserved word in C#, and we should NEVER use reserved words when naming types, objects or parameters
- you don't always persist an enum as an `int`
- there's no requirement to have a constructor with all the properties in it, in fact, that's something you would always want to avoid because every time you add or change something in your model, you would have to also change the constructor adding unnecessary maintenance effort 
- regarding the last issue you posted: do you have a default parameterless constructor in the Car class?

side note: feel free  to open a PR with the upcoming changes so we can discuss things there directly looking at the code!

**Comment by gianpieryup** [comment]
Por como no pintaba la palabra type pense que no era reservada le cambie el nombre a kind, tambien lei que puedes referenciarlas como palabras string con el modificador @type

**Comment by jrocaspana** [comment]
> Por como no pintaba la palabra type pense que no era reservada le cambie el nombre a kind, tambien lei que puedes referenciarlas como palabras string con el modificador "prefijo @" ejemplo @type

you are right, the correct keyword is `typeof`, nevertheless let's pick a better name for that enum and also let's respect code conventions.

**Comment by gianpieryup** [comment]
let's code in english! ✅ Soy peor que A1 pero me esfuerzo

let's try to manipulate a collection of cars instead of the JSON file on each method (meaning not having to deal with the index of the element). As an example: you can possibly load all the cars in memory when the CRUD class is instantiated and persist the changes on demand. ❎ La solucion que yo presente nace de la idea de que por cada peticion del CRUD cuando conectemos a una BD se tendria que hacer una nueva peticion a la BD hacerlo en memoria seria mas eficiente pero no tendria similitud y se quedaria atado a una aplicación de consola, pero si es algo que piden porque quizá este proyecto se piensa que seria mejor asi lo aceptare

let's make the Car class have its own file (not inside Program.cs)
instead of using an INT to indicate whether a car is manual or not, let's try to use an enum.
try to use implicit types when possible (remember to follow code conventions)
❎ Hay veces que me anda y veces que no, yo la verdad no entiendo ☹

**Comment by gianpieryup** [comment]
Ai pude solucionar todo, fui al commit anterior donde todo funcionaba bien y trabaje desde ahi.

**Comment by jrocaspana** [comment]
bien Giampier! vamos en castellano entonces así no agregamos complejidad innecesaria con el idioma.
lo estas haciendo muy muy  bien! te dejo algunas observaciones y comentarios debajo que creo podemos ir puliendo juntos:
- antes de hacer los próximos cambios, te invito a crear una rama nueva (branch) y luego crear un pull request con los cambios. 
- tema configuracion: casi lo tenes, te dejo de referencia algo que hizo Manuel y creo que puede inspirarte: https://github.com/manu3210/MSAcademy/pull/1/files 
- tener mas consistencia en el uso del `var` (siempre siguiendo las convenciones de código).
- no temas crear un archivo nuevo solo para poner una clase nueva que tiene 3 lineas (eg. `Configuraciones`). 
- intenta prestarle mayor atencion al casing (cuando usar mayusculas y cuando minusculas) de los nombres de las clases, tipos, propiedades y metodos.
- si podes estaria buenisimo que manipules el archivo JSON usando un bloque using 

te dejo algunos links de referencia:
https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

espero con ganas tu PR para seguir trabajando ahi! te mando un abrazo
